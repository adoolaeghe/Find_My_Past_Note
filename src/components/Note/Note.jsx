import React, { Component } from 'react'
import Modal from '../modal/modal'
import '../../style/Note.css'
import PropTypes from 'prop-types'


class Note extends Component {
  constructor(props){
    super(props);
    this.noteContent = props.noteContent;
    this.noteTitle = props.noteTitle;
    this.onClick = this.onClick.bind(this)
  }

  onClick() {
    document.getElementById(this.props.noteId.toString()).style.display = "block";
  }

  render(props){
    return(
      <div className = 'noteWrapper'>
        <div className = 'note fade-in' >
          <p className = 'noteContent'>{this.noteTitle}</p>
          <div className= 'noteTime'>Date: {this.props.date.substring(0,10)} <br/> at {this.props.date.substring(11,16)} </div>
          <button className = 'btnModal' onClick = {this.onClick.bind(this)} id = "myBtn">Open</button>
        </div>
        <Modal noteTitle = {this.noteTitle}
               noteContent = {this.noteContent}
               noteId = {this.props.noteId} />
      </div>
    )
  }
}

Note.propTypes = {
  noteContent: PropTypes.string
}

export default Note;
