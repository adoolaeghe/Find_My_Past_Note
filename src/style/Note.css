.noteWrapper{
  height: 400px;
  margin: 40px;
}

.note {
  flex: 0 0 100%;
  padding:10px 10px 38px 34px;
  height: 300px;
  margin: auto;
  margin-bottom: 8px;
  width: auto;
  max-width: 300Px;
  min-width: 300px;
  font-family: 'Lato', sans-serif;
  font-size: 1.4em;
  color: #222;
  background-color: #FFFFE0;
  border-radius: 3px;
  opacity: 1;
}

.fade-in {
    opacity:0;
    animation:fadeIn ease-in 1;
    animation-fill-mode:forwards;
    animation-duration: 0.1s;
}

.noteContent{
  float:left;
  padding-right: 70px;
  padding-bottom: 10px;
  display: block;
}

.closebtn {
    color: #AA9;
    font-weight: bold;
    float: right;
    cursor: pointer;
    font-size: 1.3em;
}

.closebtn:hover {
    color: #AC2500;
}

@keyframes fadeIn { from { opacity:0; } to { opacity:0.8; } }
@keyframes fadeOut { from { opactiy:0.8; } to { opacity:0; } }
