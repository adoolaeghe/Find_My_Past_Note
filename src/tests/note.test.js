import React from 'react';
import Note from '../components/Note/Note';
import sinon from 'sinon'
import { shallow, mount, render } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('<Note />', () => {

  const minProps = {
    noteContent: 'a new note',
    noteId: "1",
    noteTitle: 'a new note title',
    date: 'a new note date',
    onClick: () => {},
  }

  it('renders 1 <Note /> component with appropriate element', () => {
    const component = shallow(<Note {...minProps}/>);
    expect(component).toHaveLength(1);
    expect(component.find('p.noteContent').length).toEqual(1);
    expect(component.find('div.noteTime').length).toEqual(1);
    expect(component.find('button.btnModal').length).toEqual(1);
  })

  it('receive the props', () => {
    const component = shallow(
      <Note  {...minProps}/>
    );
    expect(component).toMatchSnapshot();
  });

  it('renders element according to props correctly', () => {
    const component = shallow(<Note  {...minProps}/>);
    expect(component.instance().props.noteContent).toBe('a new note');
    expect(component.instance().props.noteTitle).toBe('a new note title');
    expect(component.instance().props.date).toBe('a new note date');
  })
})
