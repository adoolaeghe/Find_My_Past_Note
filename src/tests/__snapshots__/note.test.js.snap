// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Note /> receive the props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Note
    date="a new note date"
    noteContent="a new note"
    noteId="1"
    noteTitle="a new note title"
    onClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="note fade-in"
        >
          <p
            className="noteContent"
          >
            a new note title
          </p>
          <div
            className="noteTime"
          >
            Date: 
            a new note
             
            <br />
             at 
            date
             
          </div>
          <button
            className="btnModal"
            id="myBtn"
            onClick={[Function]}
          >
            Open
          </button>
        </div>,
        <Modal
          noteContent="a new note"
          noteId="1"
          noteTitle="a new note title"
        />,
      ],
      "className": "noteWrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="noteContent"
            >
              a new note title
            </p>,
            <div
              className="noteTime"
            >
              Date: 
              a new note
               
              <br />
               at 
              date
               
            </div>,
            <button
              className="btnModal"
              id="myBtn"
              onClick={[Function]}
            >
              Open
            </button>,
          ],
          "className": "note fade-in",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "a new note title",
              "className": "noteContent",
            },
            "ref": null,
            "rendered": "a new note title",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Date: ",
                "a new note",
                " ",
                <br />,
                " at ",
                "date",
                " ",
              ],
              "className": "noteTime",
            },
            "ref": null,
            "rendered": Array [
              "Date: ",
              "a new note",
              " ",
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              " at ",
              "date",
              " ",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Open",
              "className": "btnModal",
              "id": "myBtn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Open",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "noteContent": "a new note",
          "noteId": "1",
          "noteTitle": "a new note title",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="note fade-in"
          >
            <p
              className="noteContent"
            >
              a new note title
            </p>
            <div
              className="noteTime"
            >
              Date: 
              a new note
               
              <br />
               at 
              date
               
            </div>
            <button
              className="btnModal"
              id="myBtn"
              onClick={[Function]}
            >
              Open
            </button>
          </div>,
          <Modal
            noteContent="a new note"
            noteId="1"
            noteTitle="a new note title"
          />,
        ],
        "className": "noteWrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="noteContent"
              >
                a new note title
              </p>,
              <div
                className="noteTime"
              >
                Date: 
                a new note
                 
                <br />
                 at 
                date
                 
              </div>,
              <button
                className="btnModal"
                id="myBtn"
                onClick={[Function]}
              >
                Open
              </button>,
            ],
            "className": "note fade-in",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "a new note title",
                "className": "noteContent",
              },
              "ref": null,
              "rendered": "a new note title",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Date: ",
                  "a new note",
                  " ",
                  <br />,
                  " at ",
                  "date",
                  " ",
                ],
                "className": "noteTime",
              },
              "ref": null,
              "rendered": Array [
                "Date: ",
                "a new note",
                " ",
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                " at ",
                "date",
                " ",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Open",
                "className": "btnModal",
                "id": "myBtn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Open",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "noteContent": "a new note",
            "noteId": "1",
            "noteTitle": "a new note title",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
